src/main/anovos/data_analyzer/quality_checker.py

```diff
@@ -721,185 +713,283 @@ def outlier_detection(
+            lower = sorted(
+                [i for i in [x[0], y[0], z[0]] if i is not None], reverse=True
+            )[n - 1]
+            upper = sorted([i for i in [x[1], y[1], z[1]] if i is not None])[n - 1]
+            if detection_side == "lower":
+                param = [lower, None]
+            elif detection_side == "upper":
+                param = [None, upper]
+            else:
+                param = [lower, upper]
+            params.append(param)
 
         # Saving model File if required
         if model_path != "NA":
+            if detection_side == "lower":
+                skewed_param = ["skewed_attribute", None]
+            elif detection_side == "upper":
+                skewed_param = [None, "skewed_attribute"]
+            else:
+                skewed_param = ["skewed_attribute", "skewed_attribute"]
+
+            schema = T.StructType(
+                [
+                    T.StructField("attribute", T.StringType(), True),
+                    T.StructField("parameters", T.ArrayType(T.StringType()), True),
+                ]
+            )
             df_model = spark.createDataFrame(
-                zip(list_of_cols, params), schema=["attribute", "parameters"]
+                zip(
+                    list_of_cols + skewed_cols,
+                    params + [skewed_param] * len(skewed_cols),
+                ),
+                schema=schema,
             )
             df_model.coalesce(1).write.parquet(
                 model_path + "/outlier_numcols", mode="overwrite"
             )
 
-    for i, j in zip(recast_cols, recast_type):
-        idf = idf.withColumn(i, F.col(i).cast(j))
-
-    def composite_outlier(*v):
-        output = []
-        for idx, e in enumerate(v):
-            if e is None:
-                output.append(None)
-                continue
-            if detection_side in ("upper", "both"):
-                if e > params[idx][1]:
-                    output.append(1)
-                    continue
+    def composite_outlier_pandas(col_param):
+        def inner(v):
             if detection_side in ("lower", "both"):
-                if e < params[idx][0]:
-                    output.append(-1)
-                    continue
-            output.append(0)
-        return output
+                lower_v = ((v - col_param[0]) < 0).replace(True, -1).replace(False, 0)
+            if detection_side in ("upper", "both"):
+                upper_v = ((v - col_param[1]) > 0).replace(True, 1).replace(False, 0)
 
-    f_composite_outlier = F.udf(composite_outlier, T.ArrayType(T.IntegerType()))
+            if detection_side == "upper":
+                return upper_v
+            elif detection_side == "lower":
+                return lower_v
+            else:
+                return lower_v + upper_v
+
+        return inner
 
-    odf = idf.withColumn("outliered", f_composite_outlier(*list_of_cols))
+    odf = idf
     odf.persist()
```

Comment:
In general, `persist` means that `odf` will be cached to Memory and Disk. In general, it is not recommended to do `persist` inside a function without using `unpersist`, because it can cause issues after finishing the function, as the `odf` will be on users' memory/disk even after the function finishes without their acknowledgement. I would recommend to completely remove the `persist` in here, to avoid issues in the future.

Reply:
Sure, I have removed this line.